backend/contracts/GoalNFT.sol

```diff
@@ -13,31 +13,30 @@ contract GoalNFT is ERC721URIStorage, Ownable {
     Counters.Counter private _tokenIds;
     string public baseUrl;
 
-    constructor(string memory _baseUrl) public ERC721("HabitCult", "CULT") {
+    constructor(string memory _baseUrl) ERC721("HabitCult", "CULT") {
         baseUrl = _baseUrl;
     }
 
-    function setBaseUrl(string memory _baseUrl) public returns (bool) {
+    function setBaseUrl(string calldata _baseUrl) public onlyOwner returns (bool) {
         baseUrl = _baseUrl;
         return true;
     }
 
 
-    function mintNFT(address[] memory recipients, string memory tokenURI)
-        public onlyOwner
-        returns (uint256)
-    {
-        _tokenIds.increment();
-
-        uint256 newItemId = _tokenIds.current();
-        // string memory tokenURI = string(abi.encodePacked(baseUrl, Strings.toString(newItemId)));
+    function mintNFT(address[] calldata recipients, string calldata tokenURI) public onlyOwner {
 
         for (uint i=0; i<recipients.length; i++) {
+            _tokenIds.increment();
+
+            uint256 newItemId = _tokenIds.current();
             address recipient = recipients[i];
+
+            // string memory tokenURI = string(abi.encodePacked(baseUrl, Strings.toString(newItemId)));
+
             _mint(recipient, newItemId);
             _setTokenURI(newItemId, tokenURI);
         }
 
-        return newItemId;
+            // return newItemId;
     }
 }
```

Comment:
you can't mint multiple nfts with same token id, so moved the whole part into the loop! 

not sure if you need the return statement tho... are you using it somewhere?